src/kernel.ts

```diff
@@ -373,20 +384,22 @@ class Kernel {
       }
       validateKernelId(success.data);
       this.connect(success.data);
-      return success.data;
     }, (error: utils.IAjaxError) => {
       this._onError(error);
+    }).catch((data?: any) => {
+      this._promiseDelegate.reject(data);
     });
+    return this.onReady;
   }
 
 
   /**
    * DELETE /api/kernels/[:kernel_id]
    *
-   * Shut down a kernel. Note: if useing a session, Session.shutdown()
+   * Shut down a kernel. Note: if using a session, Session.dispose()
    * should be used instead.
    */
-  shutdown(): Promise<void> {
+  dispose(): Promise<void> {
     this._handleStatus('shutdown');
     this.disconnect();
     return utils.ajaxRequest(this._kernelUrl, {
```

Comment:
Erm, I think `shutdown` was named fine. It was specifically `delete` I felt was named wrong. If `delete` never actually existed, then this is a non-issue.

I would not expect `dispose` to return a value, so actually prefer `shutdown`.